Add plugin: Vault Stats

blueheron786 opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/blueheron786/obsidian-stats-plugin/


## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

blueheron786 commented:
@Zachatoo @joethei I've been trying to get this merged in for a month (one of two plugins), any pointers on where I'm going wrong?

Zachatoo commented:
["author": "Obsidian",](https://github.com/blueheron786/obsidian-stats-plugin/blob/d805d47c59b71886f17a231d7f49d6704f29b7bd/manifest.json#L7)
Please update this to your name or username.

[const content = await this.app.vault.read(file);](https://github.com/blueheron786/obsidian-stats-plugin/blob/d805d47c59b71886f17a231d7f49d6704f29b7bd/main.ts#L20)
[Prefer to use `Vault.cachedRead` over `Vault.read` when you only want to read a file and not modify it](https://docs.obsidian.md/Plugins/Vault#Read+files)

---

You have to loop over every file in the vault 3 times in your `showStats` method, consider only looping once for better performance.

joethei commented:
https://github.com/blueheron786/obsidian-stats-plugin/blob/f13125d597cb27e06fa8690676830205e1677b67/manifest.json#L5
There is no 1.18.0 of Obsidian.
The app checks the minAppVersion, and if that is equal or lower than the currently installed version it uses the `version`, if it's not then it checks `versions.json`, which has the same problem in your case.

blueheron786 commented:
Thank you for the prompt feedback. Seems like something we can add to the automated checks. I'll target 1.8.3 instead.

joethei commented:
You don't need to change anything on this end, you just need to modify the manifest.json and versions.json in your repo.

blueheron786 commented:
OK, sounds good, thanks for all the info. I updated both files and I still see the problem after restarting Obsidian, so maybe it's an issue with caching somewhere. I'll wait and see what happens in a couple of hours.

blueheron786 commented:
I deleted versions.json since I only have one version, and updated the manifest.json version to 1.18.10, which is what I have installed. Same error. 😔
